src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -854,10 +881,10 @@ public bool Remove(TKey key, [MaybeNullWhen(false)] out TValue value)
 
         public bool TryGetValue(TKey key, [MaybeNullWhen(false)] out TValue value)
         {
-            int i = FindEntry(key);
-            if (i >= 0)
+            ref TValue entry = ref FindValueOrNull(key);
```

Comment:
The `entry` name is a little misleading, because it is not a reference to an `Entry` instance.